### DISCUSSION TITLE
Tuple Literals for parameter defaults

### DISCUSSION THREAD

**Discussion by KathleenDollard** [dotnet/csharplang#1976] [Language Ideas]
[This issue](https://github.com/dotnet/vblang/issues/353) came up in VBLang. It seems like an interesting idea, so I wanted it to appear here as well (since we'd do it both languages, if we did it).

**Comment by HaloFour** [comment]
> since we'd do it both languages

Would that necessarily be true?  C# doesn't support DateTime defaults, for example.

How would this be encoded?  Multiple attributes?  What would their values be?  What about types that VB.NET allows to be defaulted but C# does not, like `DateTime`?

**Comment by gafter** [comment]
@HaloFour C# gives no special meaning to the type `DateTime`, while VB does. That is why they handle them differently. But both language recognize tuple types as special, which is why any additional support for them is likely to show up in both languages.

**Comment by HaloFour** [comment]
@gafter 

Definitely.  And I'm not arguing against doing that.

I'm still curious about implementation ideas and what will happen when VB.NET declares a method with an optional/default tuple parameter with Dates. 😉
